core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -269,6 +269,20 @@ public <T extends RESTResponse> T post(
     return execute(Method.POST, path, null, body, responseType, headers, errorHandler);
   }
 
+  /**
+   * @deprecated Will be removed in 1.2.0, use {@link RESTClient#delete(String, Map, Class, Map,
+   *     Consumer)}.
+   */
+  @Deprecated
+  @Override
+  public <T extends RESTResponse> T delete(
```

Comment:
Do we need to deprecate this? It seems like a reasonable thing to just keep it.

Reply:
I have removed the deprecation for now, but I would still consider deprecating it. We have 4 different `delete` methods in `RESTClient`. I think just keeping the one with `queryParams` would be ok